### PR TITLE
Chunk server Implementation

### PR DESCRIPTION
**Pull request by soumith2105**
Chunk server Implementation where chunks will be saved in filesystem
- Chunks will be stored in the file system.
- They will be saved under the server port to keep them separate when multiple server instances are running.
- Chunks will also be grouped by client IP to organize them when requests come from different clients.

### CODE REVIEW

**Review by triptighanghas — DISMISSED** [review]

File: .idea/uiDesigner.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nsalazar0429) [inline .idea/uiDesigner.xml:1]: This file should not be commited , right ?

File: chunkmaster/src/main/java/com/example/chunkmaster/ChunkServerHeartbeat.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.example.chunkmaster;
+
+import java.util.List;
+import java.util.Map;
+
+public class ChunkServerHeartbeat {
```
Comment (nsalazar0429) [inline chunkmaster/src/main/java/com/example/chunkmaster/ChunkServerHeartbeat.java:6]: use same model as https://github.com/CMPE275-Distributed-Systems/CMPE_275_Final/pull/8

File: chunkmaster/src/main/java/com/example/chunkmaster/HeartbeatController.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/chunkMaster")
+public class HeartbeatController {
+    @PostMapping("/chunkServerHeartBeat")
+    public void handleChunkServerHeartBeat(@RequestBody ChunkServerHeartbeat chunkServerHeartbeat){
```
Comment (nsalazar0429) [inline chunkmaster/src/main/java/com/example/chunkmaster/HeartbeatController.java:12]: Is this receiving a heartbeat from ?

**Review by nsalazar0429 — COMMENTED** [review]
What's the difference between this PR and https://github.com/CMPE275-Distributed-Systems/CMPE_275_Final/pull/7

**Review by karandesai-sjsu — COMMENTED** [review]
few changes need in model to work with master code from the chunkmaster group , otherwise we have to go with the chunkmaster implementation this pr has

**Comment by soumith2105** [comment]
> What's the difference between this PR and #7

We pulled those changes into this branch.

File: chunkmaster/src/main/java/com/example/chunkmaster/HeartbeatController.java (hunk 3 above)
Comment (triptighanghas) [reply chunkmaster/src/main/java/com/example/chunkmaster/HeartbeatController.java:12]: this was added as an empty method, to test heartbeat endpoint, that request sent by chunkserver is received by chunkmaster.

File: chunkmaster/src/main/java/com/example/chunkmaster/ChunkServerHeartbeat.java (hunk 2 above)
Comment (triptighanghas) [reply chunkmaster/src/main/java/com/example/chunkmaster/ChunkServerHeartbeat.java]: done

**Comment by nsalazar0429** [comment]
Updates: 

- [ ] Remove initial read since we won't support it

**Review by nsalazar0429 — DISMISSED** [review]

**Review by adityarao97 — APPROVED** [review]

**Review by triptighanghas — APPROVED** [review]

**Pull request merged**